initial implementation for addon framework

ibrokethecloud opened the pull request:
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
<!-- Explain the problem you are aiming to resolve in this PR. -->
The PR introduces an Addon CRD and controller, which wraps around the HelmChart CRD.
This allows new optional components to be packaged with harvester, while not enabling them.
This ensure that the core harvester requirements stay within reasonable bounds.

In addition, this also allows us to move off ManagedCharts and eventually HelmCharts as the means to package components.

**Solution:**
<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

**Related Issue:**

**Test plan:**
<!-- Make sure tests pass on the Circle CI. -->
*Enable vm-import-controller add-on*
- No addon is enabled by default, you can check with the command.
```
kubectl get addons.harvesterhci.io -A
```
- Enable the `vm-import-controller` addon:
```
kubectl edit addons.harvesterhci.io vm-import-controller -n harvester-system
```
And change `spec.enabled` to `true`.
- You should see a vm-import-controller HelmChart CR:
```
$ kubectl get helmchart -n harvester-system
NAME                   AGE
vm-import-controller   2s
```
- And a pod
```
$ kubectl get pods -n harvester-system | grep vm-import
helm-install-vm-import-controller-k5hm9                           0/1     Completed   0             61s
vm-import-controller-harvester-vm-import-controller-84749cns69l   1/1     Running     0             58s
```

*Disable vm-import-controller add-on*

- The add-on can also be disabled:
```
kubectl edit addons.harvesterhci.io vm-import-controller -n harvester-system
```
And change `spec.enabled` to `false`.

- You should see a vm-import-controller HelmChart CR is gone:
```
$  kubectl get helmchart -n harvester-system
No resources found in harvester-system namespace.
```

- And no vm-import-controller pod:
```
kubectl get pods -n harvester-system | grep vm-import
```

ibrokethecloud committed:
merge fix

guangbochen reviewed (commented):
In general, this is LGTM, we can prepare to squish the commits where applicable, thanks.

ibrokethecloud committed:
change version check

ibrokethecloud committed:
changed version check for upgrade

ibrokethecloud committed:
Added addons to upgrade image
changed lib.sh to use absolute path to addons

updated addon path during version check

minor changes

bk201 reviewed (approved):
LGTM.

w13915984028 reviewed (approved):
LGTM, thanks.
